pages/user-interfaces.html

```diff
@@ -55,102 +43,119 @@
+</p>
+
+<p>
+As a way to try Coq without installing anything, you can also
+use <a href="https://jscoq.github.io/">JsCoq</a> which loads Coq
+entirely in <strong>your browser</strong>.  However, it is limited to conduct
+actual projects: e.g., it lacks access to the VM and native computing
+machineries of Coq, and running in the browser is slower.
 </p>
 
 </div>
 
 <div class="frameworklinks">
 <ul>
-<li><a class="extlink" href="https://github.com/coq-community/vscoq">VsCoq (VS Code)</a></li>
-<li><a class="extlink" href="https://github.com/ejgallego/coq-lsp">coq-lsp (VS Code)</a></li>
-<li><a class="extlink" href="https://proofgeneral.github.io/">Proof General (Emacs)</a></li>
-<li><a class="extlink" href="https://github.com/whonore/Coqtail">Coqtail (Vim)</a></li>
+<li><a class="extlink" href="https://github.com/coq-community/vscoq/tree/vscoq1">VsCoq 1</a></li>
+<li><a class="extlink" href="https://proofgeneral.github.io/">Proof General</a></li>
+<li><a class="extlink" href="https://github.com/whonore/Coqtail">Coqtail</a></li>
+<li><a href="refman/practical-tools/coqide.html">CoqIDE</a></li>
+<li><a class="extlink" href="https://jscoq.github.io/">JsCoq</a></li>
 </ul>
 </div>
 </div>
 
 <div class="framework">
-<div class="frameworklabel">Standalone interfaces</div>
+<div class="frameworklabel">Towards an LSP-based modular design of Coq user interfaces</div>
 <div class="frameworkcontent">
-
-<p>
-Alternatively, you can
-use <a href="https://coq.inria.fr/refman/practical-tools/coqide.html">CoqIDE</a>,
-which is developed and distributed alongside Coq. CoqIDE relies
-on <a href="https://github.com/coq/coq/blob/master/dev/doc/xml-protocol.md">Coq's
-XML protocol for IDEs</a> and implements all of the features provided
-by this protocol, meaning in particular asynchronous evaluation.
-CoqIDE comes with standard basic Emacs-like bindings
-and with some specific Coq-related bindings but, as an editor, it is
-not as complete as a general-purpose editor could be (for instance it
-does not have automatic indentation).
-</p>
-
-<p>
-As a way to try Coq without installing anything, you can
-use <a href="https://jscoq.github.io/">JsCoq</a>.  JsCoq loads Coq
-entirely in your browser.  However, it is too limited to conduct
-actual projects: it lacks access to the VM and native computing
-machineries of Coq, and may hit out-of-memory and stack-overflow
-failures quicker than native versions of Coq.
-</p>
-
+  The <a href="https://microsoft.github.io/language-server-protocol">Language
+        Server Protocol</a> (LSP) is a standard protocol for interfacing
+        IDEs and programming language tools. Two interfaces, each composed of a VS Code
+        client and of a Coq server communicating over a proof-assistant-specific
+        extension of LSP are at an experimental stage of development:
+  <ul>
+    <li><a href="https://github.com/coq-community/vscoq">VsCoq 2</a></li>
+    <li><a href="https://github.com/ejgallego/coq-lsp">coq-lsp</a></li>
+  </ul>
```

Comment:
For consistency with the formatting of the rest of the page (and also probably more correct HTML). I'm only adding the `<p>` / `</p>` in the beginning and at the end.
```suggestion
  <p>
  The <a href="https://microsoft.github.io/language-server-protocol">Language
        Server Protocol</a> (LSP) is a standard protocol for interfacing
        IDEs and programming language tools. Two interfaces, each composed of a VS Code
        client and of a Coq server communicating over a proof-assistant-specific
        extension of LSP are at an experimental stage of development:
  <ul>
    <li><a href="https://github.com/coq-community/vscoq">VsCoq 2</a></li>
    <li><a href="https://github.com/ejgallego/coq-lsp">coq-lsp</a></li>
  </ul>
  </p>
```